hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -399,4 +775,116 @@ public static int mapRecordKeyToFileGroupIndex(String recordKey, int numFileGrou
     return fileSliceStream.sorted((s1, s2) -> s1.getFileId().compareTo(s2.getFileId())).collect(Collectors.toList());
   }
 
+  public static List<HoodieRecord> convertMetadataToColumnStatsRecords(HoodieCommitMetadata commitMetadata,
```

Comment:
comment about L760 to L767. I see we are instantiating the HoodieTableFileSystemView everytime here. May be we can try to move the fsView instantiation to the caller and thus amortize the cost. For eg, 
```
enablePartition(MetadataPartitionType.FILES, metadataConfig, metaClient, isBootstrapCompleted);
```
we make similar calls thrice with this patch. So, we can instantiate just once outside and use the same here. Try to check for other callers too if we can do any such amortization.

Reply:
Refactored this and made use of a single FSView